zingolib/src/lightclient/sync.rs

```diff
@@ -151,13 +151,13 @@ impl LightClient {
     }
 
     /// TODO: Add Doc Comment Here!
-    pub fn start_mempool_monitor(lc: Arc<LightClient>) {
+    pub fn start_mempool_monitor(lc: Arc<LightClient>) -> Result<(), ()> {
```

Comment:
Why not define a proper Error here?

Reply:
returning a Result narrowed the possible causes for a different error

Reply:
Why a Result with `()` as the `Error`?   Why not define a `thiserror` attributed enum with error messages?